openshift/template-mongodb-backup.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: mongodb-backup-template
+  annotations:
+    description: 'Template for MongoDB container backup job'
+    tags: 'database,backup,mongo'
+parameters:
+  - name: MONGODB_BACKUP_KEEP
+    description: 'Number of backups to keep'
+    value: '24'
+  - name: MONGODB_BACKUP_SCHEDULE
+    description: 'Cron-like schedule expression. Default: Every hour at :50'
+    value: '50 * * * *'
+  - name: MONGODB_BACKUP_VOLUME_CLAIM
+    description: 'Name of the volume claim to be used as storage'
+    value: 'mongodb-backup'
+  - name: MONGODB_ADMIN_PASSWORD
+    description: 'MongoDB admin user''s password. Required.'
+objects:
+  - apiVersion: batch/v1beta1
+    kind: CronJob
+    metadata:
+      name: mongodb-backup
+    spec:
+      schedule: ${MONGODB_BACKUP_SCHEDULE}
+      concurrencyPolicy: Forbid
+      jobTemplate:             
+        spec:
+          template:
+            spec:
+              volumes:
+                - name: mongodb-backup
+                  persistentVolumeClaim:
+                    claimName: ${MONGODB_BACKUP_VOLUME_CLAIM}
+              containers:
+                - name: mongodb-backup
+                  image: 'docker-registry.default.svc:5000/openshift/mongodb:latest'
```

Comment:
can we use the same version of mongo we are using for our replica set?

Reply:
Makes sense. Fixed it.